src/MusicStore/test/MusicStore.E2ETests/MusicStore.E2ETests.csproj

```diff
@@ -39,4 +36,40 @@
     <Reference Include="System.Data.SqlClient" />
   </ItemGroup>
 
+  <Target Name="_RunPublishAssets" BeforeTargets="AfterBuild" DependsOnTargets="PublishAssets" Condition="'$(SkipTests)' != 'true' OR '$(BuildHelixPayload)' != 'false'" />
+
+  <Target Name="PublishAssets">
+    <ItemGroup>
+      <Architecture Include="x64" />
+      <Architecture Include="x86" Condition="$([MSBuild]::IsOSPlatform('Windows'))" />
+      <ApplicationType Include="Portable;Standalone" Architecture="%(Architecture.Identity)" />
+
+      <PublishAsset
+        Include="..\..\samples\MusicStore\MusicStore.csproj"
+        Architecture="%(ApplicationType.Architecture)"
+        ApplicationType="%(ApplicationType.Identity)"
+        PublishDirectory="$(MSBuildThisFileDirectory)$(OutputPath)testassets\%(ApplicationType.Identity)\%(ApplicationType.Architecture)\" />
+    </ItemGroup>
+
+    <MSBuild
+      Projects="$(MSBuildProjectFullPath)"
+      Targets="_PublishAsset"
+      Properties="PublishProject=%(PublishAsset.Identity);Architecture=%(Architecture);ApplicationType=%(ApplicationType);PublishDirectory=%(PublishDirectory)"
+      BuildInParallel="false" />
+  </Target>
+
+  <Target Name="_PublishAsset">
```

Comment:
Did you try reusing targets from testsite.props?

Reply:
NIH bro. But more seriously, the IIS targets had a bunch of things that look like they are very IIS specific (copying launch settings \ injecting deps file etc). All we need here is to publish a project with different settings. I did re-use the `PublishedApplicationPublisher` which majorly speeds up things.

Reply:
I think it be easily extracted, you mostly need https://github.com/aspnet/AspNetCore/blob/master/src/Servers/IIS/build/testsite.props#L75-L113. There are couple workarounds for VS/publish issues there that are universaly useful.